src/sortStudents.ts

```diff
@@ -1,16 +1,92 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function createFilterCallbackMarried(order: SortOrder):
+(a: Student, b: Student) => number {
+  if (order === 'asc') {
+    return (studentA: { married: number }, studentB: {married: number}):
+    number => studentA.married - studentB.married;
+  }
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return (studentA: { married: number }, studentB: {married: number}):
+  number => studentB.married - studentA.married;
+}
+
+function createFilterCallback(order: SortOrder, sortBy: SortType):
+(a: Student, b: Student) => number {
+  if (order === 'asc') {
+    return (studentA: { sortBy: SortType }, studentB: { sortBy: SortType }):
+    number => studentA[sortBy].localeCompare(studentB[sortBy]);
+  }
+
+  return (studentA: { age: number }, studentB: {age: number}):
+  number => studentB[sortBy].localeCompare(studentA[sortBy]);
+}
```

Comment:
```suggestion
function createFilterCallback(
  order: SortOrder,
  sortBy: SortType,
): (a: Student, b: Student) => number {
  if (order === 'asc') {
    return (studentA: { sortBy: SortType }, studentB: { sortBy: SortType }):
      number => studentA[sortBy].localeCompare(studentB[sortBy]);
  }

  return (studentA: { age: number }, studentB: {age: number}):
    number => studentB[sortBy].localeCompare(studentA[sortBy]);
}
```
pay attention to the code style, fix it all over the code